src/demo-app/datepicker/datepicker-demo.ts

```diff
@@ -40,4 +41,49 @@ export class DatepickerDemo {
 
   onDateInput = (e: MatDatepickerInputEvent<Date>) => this.lastDateInput = e.value;
   onDateChange = (e: MatDatepickerInputEvent<Date>) => this.lastDateChange = e.value;
+
+  // pass custom header component type as input
+  customHeader = CustomHeader;
+}
+
+// Custom header component for datepicker
+@Component({
+  selector: 'custom-header',
+  template: `
+      <div>
+        <button (click)="previousClicked('year')">&lt;&lt;</button>
+          <button (click)="previousClicked('month')">&lt;</button>
+        {{periodLabel}}
+        <button (click)="nextClicked('month')">&gt;</button>
+          <button (click)="nextClicked('year')">&gt;&gt;</button>
+      </div>
+  `
+})
+export class CustomHeader {
+  constructor(@Host() public calendar: MatCalendar<any>,
```

Comment:
might as well make calendar private too, since it doesn't need to be public

Reply:
done in c7f21bfa6cf06845dbce6a41049890637d150fb7